clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java

```diff
@@ -0,0 +1,285 @@
+   */
+  public FileOutStream getOutStream(TachyonURI path, ClientOptions options) throws IOException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      long fileId = masterClient.createFile(path.getPath(), options.getBlockSize(), true);
+      return new FileOutStream(fileId, options);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  // TODO: We should remove this when the TachyonFS code is fully deprecated
+  @Deprecated
+  public FileOutStream getOutStream(long fileId, ClientOptions options) throws IOException {
+    return new FileOutStream(fileId, options);
+  }
+
+  /**
+   * If the file is a folder, return the {@link FileInfo} of all the direct entries in it.
+   * Otherwise return the FileInfo for the file. The file infos are snapshots of the
+   * file metadata, and the locations, last modified time, and path are possibly inconsistent.
+   *
+   * @param file the handler for the file
+   * @return a list of FileInfos representing the files which are children of the given file
+   * @throws IOException if the file does not exist
+   */
+  @Override
+  public List<FileInfo> listStatus(TachyonFile file) throws IOException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      return masterClient.getFileInfoList(file.getFileId());
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Adds metadata about a file in the under storage system to Tachyon. Only metadata will be
+   * updated and no data will be transferred. The data can be added to Tachyon space by doing an
+   * operation with the cache option specified, for example reading.
+   *
+   * @param path the path to create the file in Tachyon
+   * @param ufsPath the under storage system path of the file that will back the Tachyon file
+   * @param recursive if true, the parent directories to the file in Tachyon will be created
+   * @return the file id of the resulting file in Tachyon
+   * @throws IOException if the Tachyon path is invalid or the ufsPath does not exist
+   */
+  public long loadFileFromUfs(TachyonURI path, TachyonURI ufsPath, boolean recursive) throws
+      IOException {
+    FileSystemMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      return masterClient.loadFileFromUfs(path.getPath(), ufsPath.getPath(), -1L, recursive);
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Creates a folder. If the parent folders do not exist, they will be created automatically.
+   *
+   * @param path the handler for the file
+   * @return true if the folder is created successfully or already existing, false otherwise.
+   * @throws IOException if the master cannot create the folder under the specified path
+   */
+  @Override
+  public boolean mkdirs(TachyonURI path) throws IOException {
```

Comment:
Does it support creating parent folders recursively? If not, does it throw exceptions or return false? Please be more specific in the javadoc.